persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/SignificantlyMutatedGeneSparkRepository.java

```diff
@@ -0,0 +1,80 @@
+package org.cbioportal.persistence.spark;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.cbioportal.model.MutSig;
+import org.cbioportal.model.meta.BaseMeta;
+import org.cbioportal.persistence.SignificantlyMutatedGeneRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Component
+@Qualifier("significantlyMutatedGeneSparkRepository")
+public class SignificantlyMutatedGeneSparkRepository implements SignificantlyMutatedGeneRepository {
+
+    @Autowired
+    private SparkSession spark;
+
+    @Value("${data.parquet.folder}")
+    private String PARQUET_DIR;
+
+    private static Log logger = LogFactory.getLog(SignificantlyMutatedGeneSparkRepository.class);
+
+    @Override
+    public List<MutSig> getSignificantlyMutatedGenes(String studyId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        System.out.println(PARQUET_DIR);
+        Dataset<Row> mutationDf = spark.read()
+            .parquet(PARQUET_DIR + "/" + studyId + "/data_mutations_extended.txt.parquet");
+
+        // for MSK
+        if (studyId.contains("msk")) {
```

Comment:
This is just for development, we obviously don't want to hardcode study ids into the code.

Reply:
The study Ids are not hardcoded, but the parquet filenames are. I added this because the other studies, genie and brca_tcga, didn't have data_mutations_mskcc.txt file, but msk_impact_2017 did. I thought msk uses the data_mutations_mskcc.txt file. If this is not true I can remove this..

Reply:
Maybe an enum or some static file for filenames would be cleaner than having "data_mutations_extended.txt.parquet" and so on?

Reply:
why not set [mergeSchema](https://spark.apache.org/docs/latest/sql-data-sources-parquet.html#schema-merging) option to true and read all the mutation files together. Something like
`spark.read.option("mergeSchema", "true").parquet(PARQUET_DIR + "/" + studyId + "/data_mutations*.txt")`

Reply:
thanks for the suggestion. Updated code to load data_mutations*